bill/invoice_test.go

```diff
@@ -516,19 +547,27 @@ func TestRemoveIncludedTaxDeep(t *testing.T) {
 	i2, err := i.RemoveIncludedTaxes()
 	require.NoError(t, err)
 
-	//data, _ := json.MarshalIndent(i2, "", "  ")
-	//t.Log(string(data))
-
 	assert.Empty(t, i2.Tax.PricesInclude)
 	l0 := i2.Lines[0]
-	assert.Equal(t, "48.84906", l0.Item.Price.String()) // note extra digit!
-	assert.Equal(t, "17781.05784", l0.Sum.String())
+	assert.Equal(t, "48.8491", l0.Item.Price.String()) // note extra digit!
+	assert.Equal(t, "17781.0724", l0.Sum.String())
 	l1 := i2.Lines[1]
-	assert.Equal(t, "49.13208", l1.Item.Price.String())
-	assert.Equal(t, "49.13208", l1.Sum.String())
+	assert.Equal(t, "49.1321", l1.Item.Price.String())
+	assert.Equal(t, "49.1321", l1.Sum.String())
 
-	assert.Equal(t, "17830.19", i2.Totals.Total.String())
-	assert.Equal(t, i.Totals.Total.String(), i2.Totals.Total.String())
+	data, _ := json.Marshal(i.Lines)
+	t.Logf("LINES: %v", string(data))
+	data, _ = json.Marshal(i.Totals)
+	t.Logf("TOTALS: %v", string(data))
+	data, _ = json.Marshal(i2.Lines)
+	t.Logf("Lines: %v", string(data))
+	data, _ = json.Marshal(i2.Totals)
+	t.Logf("TOTALS: %v", string(data))
+
+	assert.Equal(t, "17830.20", i2.Totals.Total.String())
+	// assert.Equal(t, i.Totals.Total.String(), i2.Totals.Total.String())
+	assert.Equal(t, "17830.20", i2.Totals.Total.String())
+	assert.Equal(t, "-0.02", i2.Totals.Rounding.String())
```

Comment:
This is a bit of a hack, but it ensures rounding errors after removing taxes are more or less handled.